pkg/network/node/pod.go

```diff
@@ -479,11 +498,16 @@ func (m *podManager) setup(req *cniserver.PodRequest) (cnitypes.Result, *running
 		}
 	}()
 
-	v1Pod, err := m.kClient.CoreV1().Pods(req.PodNamespace).Get(context.TODO(), req.PodName, metav1.GetOptions{})
-	if err != nil {
+	podLister := m.podInformers.Core().V1().Pods().Lister()
+	if err = wait.PollImmediateWithContext(context.Background(), 10*time.Millisecond, 5*time.Second, func(ctx context.Context) (bool, error) {
```

Comment:
So this is the really ugly part... we're converting the code to using informers/listers... except that semantically, this code can't actually be based on informers, because it needs to guarantee it gets the latest pod state from the apiserver, not "whatever state the informer currently knows about".

And especially, the performance problems this PR is trying to fix only exist in one weird case (upgrade requiring a bridge recreation), but the "awkward" part of this PR (looping while waiting for the informer to catch up) isn't needed in that case (all of the pods to be reattached will already have been cached by the informer), and instead will only get used in the "normal" pod startup case, where this PR doesn't actually help at all anyway. :slightly_frowning_face: 

So, basically the idea of this PR is, if we're doing a bunch of pod setups all at once, it would be a lot better to make a single API call to get all of the pod objects rather than doing a separate call for each one... maybe we should do that explicitly?

1. add a new `podManager.cachePods(pods []*corev1.Pod)` to store an array of pods
2. modify `podManager.setup` to look for the pod it wants in the pod manager's list of cached pods first, and only call `Get()` if the pod isn't there.
3. have the pod reattach code call `pods, err := common.ListPodsInNodeAndNamespace(ctx, node.kClient, node.hostName, metav1.NamespaceAll)` and then `podManager.cachePods(pods)` before it starts reattaching pods, and `podManager.cachePods(nil)` after it finishes.

I think that would end up being simpler/clearer?